### DISCUSSION TITLE
DNS for pod not working when coredns is on a different node

### DISCUSSION THREAD

**Discussion by saisujith2** [k3s-io/k3s#7402] [Q&A]
**Environmental Info:**
K3s Version: 

`
k3s version v1.27.1+k3s1 (bc5b42c2)  
go version go1.20.3
`

Node(s) CPU architecture, OS, and Version: 

`
Linux <nodename> 4.18.0-372.13.1.el8_6.x86_64 #1 SMP Mon Jun 6 15:05:22 EDT 2022 x86_64 x86_64 x86_64 GNU/Linux
`

**Cluster Configuration:**

1 Server
1 Agent

**Describe the bug:**

DNS resolution is not working on agent (node without a coredns pod). 

Ping Google.com response on server
<img width="489" alt="image" src="https://user-images.githubusercontent.com/75458037/235715498-cde48221-34e7-4367-8327-970e95e9093b.png">

Ping Google.com response on agent
<img width="270" alt="image" src="https://user-images.githubusercontent.com/75458037/235716698-8e52ba5e-c849-486c-90a5-969034cc2a39.png">

nslookup kubernetes response on server
<img width="357" alt="image" src="https://user-images.githubusercontent.com/75458037/235715626-c3b7efd8-f353-4bf2-933d-b41134fbf63c.png">

nslookup kubernetes response on agent, Ping <own_nodename> also has the same response
<img width="406" alt="image" src="https://user-images.githubusercontent.com/75458037/235716836-54bf73f0-cc9c-43e9-81fc-2662495e443e.png">

**Steps To Reproduce:**

- Installed K3s: 

**Server Installation**
sudo curl -sfL https://get.k3s.io |  INSTALL_K3S_VERSION="v1.27.1+k3s1" sh -s - server --cluster-init --config '/apps/rancher/k3s-config/config.yaml

```
config.yaml
server: "https://server_nodename.org:6443"
token: "example"
data-dir: "/apps/rancher/k3s/"
kubelet-arg:
  - "root-dir=/apps/rancher/kubelet"
```

**Agent Installation**
sudo curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION="v1.19.11+k3s1" sh -s - agent --config '/apps/rancher/k3s-config/config.yaml'

config.yaml
```
token: "example"
write-kubeconfig-mode: "0644"
data-dir: "/apps/rancher/k3s/"
kubelet-arg:
  - "root-dir=/apps/rancher/kubelet"
node-label:
  - "foo=bar"
  - "something=amazing"
kube-apiserver-arg: "event-ttl=12h0m0s"
cluster-cidr: "100.100.0.0/18"
service-cidr: "100.100.64.0/18"
disable:
  - "traefik"
```

Deployed a busybox container for testing the ping and nslookup commands
```
apiVersion: v1
kind: Pod
metadata:
  name: busybox
  namespace: default
spec:
  containers:
    - name: busybox
      image: >-
        busybox:1.35
      command:
        - /bin/sh
        - '-c'
        - sleep infinity 
```

**Expected behavior:**
It is expected that the dns resolution works on all pods in the cluster irrespective of the node they are deployed on.

**Actual behavior:**

There is no dns resolution happening on the pod that is deployed on the agent.

**Additional context / logs:**

CoreDNS logs:
```
[WARNING] No files matching import glob pattern: /etc/coredns/custom/*.server
```

Also tried the same steps with 2 servers both masters without any luck.

Any help is much appreciated.

**Comment by brandond** [comment]
This indicates that CNI traffic (pod to pod or pod to service) is being dropped when it needs to transit between nodes. Please be sure that you've opened all the correct ports to allow CNI traffic for whatever flannel backend you're using. See the port requirements at https://docs.k3s.io/installation/requirements#networking. 

Note that we generally recommend disabling any other iptables-based firewalls such as firewalld or ufw as they may interfere with the rules managed by Kubernetes components. If you want to try anyway, you can check out the suggestions at https://docs.k3s.io/advanced#red-hat-enterprise-linux--centos

**Comment by saisujith2** [comment]
I have made sure that the below ports are open for both master and node

- TCP 2379-2380
- TCP 6443
- UDP 8472
- TCP 10250

Also, firewalld & nm-cloud-setup disabled

Can I do anything else to troubleshoot this issue?
